prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1536,22 +1536,24 @@ int rxr_ep_grow_rx_pkt_pools(struct rxr_ep *ep)
 		}
 	}
 
-	assert(ep->rx_unexp_pkt_pool);
```

Comment:
I think there is still value in the `assert`, you just need to move it into the `if` block

Reply:
At the beginning, I added an `if (rxr_env.rx_copy_unexp)` and then add the assert into the `if` block. But later I found the readcopy pkt pools in this function just check `if (ep->rx_readcopy_pkt_pool)` before the grow. I could keep the assert inside the `if(ep->rx_unexp_pkt_pool)` but I am curious how assert can fail in this case?

Reply:
Assertion is used exactly for something that should never fail (unless there is bug or memory corruption). If something can fail, then it should be handling with error handling (return err code, etc).

Reply:
I see, thanks. I added back the assertions in updated PR.